lib/ansible/parsing/vault/__init__.py

```diff
@@ -1094,7 +1094,9 @@ def shuffle_files(self, src, dest):
             os.chown(dest, prev.st_uid, prev.st_gid)
 
     def _editor_shell_command(self, filename):
-        env_editor = os.environ.get('EDITOR', 'vi')
+        env_editor = os.environ.get('VISUAL')
+        if not env_editor:
+            env_editor = os.environ.get('EDITOR', 'vi')
```

Comment:
Since we have used `EDITOR` historically, we should not change the default and need to start with `EDITOR` as the default. This precedence may not make sense, though, and we may need to add a config option to control this.
```suggestion
        env_editor = os.environ.get('EDITOR', os.environ.get('VISUAL', 'vi'))
```